tests/test_dataloader.py

```diff
@@ -248,10 +249,10 @@ def test_denoising_dataloader(decoder_only_format: bool, pretokenize: bool,
 
 @pytest.mark.parametrize('decoder_only_format', [True, False])
 @pytest.mark.parametrize('allow_pad_trimming', [True, False])
-@pytest.mark.parametrize('packing_ratio', [10.0, None])
+@pytest.mark.parametrize('packing_ratio', [10.0, None, 'auto'])
 def test_finetuning_dataloader(decoder_only_format: bool,
                                allow_pad_trimming: bool,
-                               packing_ratio: Optional[float]):
+                               packing_ratio: Optional[Any]):
```

Comment:
Nit (sorry): Should the signature be `Optional[Union[int, float, str]]`?

Reply:
good point! i think it would be `Optional[Union[float, str]]`?

Reply:
It can be, but I think for typing purposes float also accepts int (even though it's not a proper subclass)

Reply:
will use `Literal['auto']`, that's better than just `str`